server/src/main/java/org/elasticsearch/cluster/routing/RoutingNode.java

```diff
@@ -371,6 +371,10 @@ public ShardRouting[] copyShards() {
         return shards.values().toArray(EMPTY_SHARD_ROUTING_ARRAY);
     }
 
+    public Index[] copyIndices() {
+        return shardsByIndex.keySet().toArray(Index.EMPTY_ARRAY);
```

Comment:
@dakrone do you know of a different/more efficient way of getting all the indices on a node?

Reply:
@dakrone I merged this PR to move forward but please if you know about a better way to get all the indices per node, I will fix it in a new PR.